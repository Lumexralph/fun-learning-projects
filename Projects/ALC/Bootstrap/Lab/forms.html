<!doctype html>
<html lang="en">

<head>
  <title>Music Album</title>
  <!-- Required meta tags -->
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <!-- Bootstrap CSS -->
  <link rel="stylesheet" href="styles/bootstrap.min.css">
  <style>
    body {
      padding-top: 70px;
    }
  </style>
</head>

<body>
  <div class="container">
      <form class="form-horizontal">
          <div class="form-group has-error has-feedback">
            <label for="address" class="control-label col-md-3">Address:</label>
            <div class="col-md-6">
              <input type="text" name="address" id="address" class="form-control" />
              <!-- Glyphicon is placed after textbox -->
              <span class="glyphicon glyphicon-remove form-control-feedback"></span>
            </div>
          </div>
         
          <!-- Input group -->
          <div class="form-group has-error has-feedback">
            <label for="username" class="control-label col-md-3">Username:</label>
            <div class="col-md-6">
              <div class="input-group">
                <div class="input-group-addon">
                  <span class="glyphicon glyphicon-user"></span>
                </div>
                <input type="text" class="form-control" name="username" id="username" />
              </div>
              <!-- Glyphicon is placed after input group-->
              <span class="glyphicon glyphicon-remove form-control-feedback"></span>
            </div>
          </div>
        </form>
        <div class="alert alert-warning alert-dismissable">
            <button type="button" class="close" data-dismiss="alert" aria-hidden="true">Ã—</button>
            <h3>Warning</h3>
            <div>It appears that something has gone wrong</div>
          </div>
  </div>
  

  <!-- Optional JavaScript -->
  <!-- jQuery first, then Bootstrap JS -->
  <script src="scripts/jquery-3.3.1.min.js"></script>
  <script src="scripts/bootstrap.min.js"></script>
</body>

</html>