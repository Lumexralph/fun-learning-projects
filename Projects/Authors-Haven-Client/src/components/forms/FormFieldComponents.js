import React from 'react';
import { Form, Label } from 'semantic-ui-react';

const FormField = (type, name, onChange, value, placeholder, label, errors) => (
  <Form.Field>
    <label htmlFor={name}>
      {label}
      <input
        required
        type={type}
        id={name}
        name={name}
        onChange={onChange}
        value={value}
        placeholder={placeholder}
      />
    </label>
    {errors && (
      <Label color="red" pointing>
        {errors}
      </Label>
    )}
  </Form.Field>
);

export default FormField;
