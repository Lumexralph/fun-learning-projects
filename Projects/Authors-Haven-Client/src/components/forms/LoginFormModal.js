import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
  Button,
  Dimmer,
  Form,
  Loader,
  Modal,
  TransitionablePortal,
  Message,
} from 'semantic-ui-react';
import ModalFormHOC, { modalOptions } from '../hoc/ModalFormHOC';
import LoginFields from './FormFieldComponents';

export const LoginForm = ({
  state,
  show,
  close,
  onSubmit,
  onChange,
}) => {
  const {
    errors,
    loading,
    open,
    user: { email, password },
  } = state;

  return (
    <div>
      <Link href="#" to="#" onClick={show}>
        Login with email
      </Link>
      <TransitionablePortal transition={modalOptions} onClose={close} open={open}>
        <Modal
          centered
          closeIcon
          id="modal-form"
          size="tiny"
          dimmer="blurring"
          open={open}
          onClose={close}
        >
          <Dimmer active={loading}>
            <Loader>Preparing your engagement</Loader>
          </Dimmer>
          <Modal.Header>Welcome</Modal.Header>
          {
            errors.randomError
            && (
            <Message negative>
              <Message.Header>{errors.randomError}</Message.Header>
              {errors.randomError === 'Your account has not been activated' ? (
                <Message.Header>
                Click
                  <Link to={`/verify-email?email=${email}`}> here </Link>
                to verify
                </Message.Header>
              ) : null}
            </Message>
            )
            }
          <Modal.Content>
            <Form className="innerForm " size="large" onSubmit={onSubmit}>
              {/* Rendering the form fields from the FormField component */}
              {LoginFields('email', 'email', onChange, email, 'example@email.com', 'Email:', errors.emailError)}
              {LoginFields('password', 'password', onChange, password, 'Enter your password', 'Password:', errors.passwordError)}
              <Button className="btn" type="submit">
                Login
              </Button>
            </Form>
          </Modal.Content>
          <Modal.Actions>
          Forgot password?
            <Link to="/reset/edit"> Recover it here.</Link>
          </Modal.Actions>
        </Modal>
      </TransitionablePortal>
    </div>
  );
};

LoginForm.propTypes = {
  state: PropTypes.shape({
    user: PropTypes.shape({
      email: PropTypes.string.isRequired,
      password: PropTypes.string.isRequired,
    }).isRequired,
    loading: PropTypes.bool.isRequired,
    open: PropTypes.bool.isRequired,
  }).isRequired,
  show: PropTypes.func.isRequired,
  close: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default ModalFormHOC(LoginForm);
