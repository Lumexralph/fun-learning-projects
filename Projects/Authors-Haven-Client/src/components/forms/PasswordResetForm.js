import React, { Component } from 'react';
import {
  Button,
  Dimmer,
  Form,
  Loader,
  Message,
  Image,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';
import FormField from './FormFieldComponents';
import RenderDiv from '../../helpers/helper';

export class ResetForm extends Component {
  state = {
    user: {
      password: '',
      confirmPassword: '',
    },
    errors: {},
    loading: false,
  };

  componentDidUpdate(prevProps) {
    const { result, error } = this.props;
    if ((result !== prevProps.result) || (error !== prevProps.error)) {
      this.updateCurrentState(false);
    }
  }

  onSubmit = (event) => {
    event.preventDefault();
    const { user } = this.state;
    const { submit } = this.props;
    const errors = this.validate(user);
    this.setState({
      errors,
    });
    if (Object.keys(errors).length === 0) {
      this.updateCurrentState(true);
      submit(user.password);
    }
  };

  onChange = (event) => {
    const { user } = this.state;
    this.setState({
      user: {
        ...user,
        [event.target.name]: event.target.value,
      },
      errors: {},
    });
  };

  validate = (data) => {
    const errors = {};
    if (data.password !== data.confirmPassword) errors.confirmPassword = 'your passwords do not match';
    return errors;
  }

  updateCurrentState(newName) {
    this.setState({
      loading: newName,
    });
  }

  render() {
    const { user, errors, loading } = this.state;
    const { error, result } = this.props;
    return (
      <div id="reset-page">
        <Dimmer active={loading}><Loader>Resetting</Loader></Dimmer>
        <div id="reset-div" className="reset-div">
          <Image id="key" src="https://image.flaticon.com/icons/svg/481/481195.svg" />
          <h2>Reset Your Password&nbsp;&nbsp;</h2>
          {result ? <Message success id="result">{result}</Message> : null}
          <Form onSubmit={this.onSubmit}>
            {error ? <Message negative>{error.body[0]}</Message> : null}
            {FormField(
              'password', 'password', this.onChange, user.password,
              'Enter new password', '',
            )}
            {FormField(
              'password', 'confirmPassword', this.onChange, user.confirmPassword,
              'Confirm password', '', errors ? errors.confirmPassword : null,
            )}
            <Button type="submit" content="Submit" />
            <RenderDiv />
          </Form>
        </div>
      </div>
    );
  }
}

ResetForm.propTypes = {
  submit: PropTypes.func.isRequired,
  error: PropTypes.string.isRequired,
  result: PropTypes.string.isRequired,
};
export default ResetForm;
