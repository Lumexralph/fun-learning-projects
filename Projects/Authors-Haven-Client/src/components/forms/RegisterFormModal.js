import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
  Button,
  Dimmer,
  Form,
  Loader,
  Modal,
  TransitionablePortal,
  Message,
} from 'semantic-ui-react';
import ModalFormHOC, { modalOptions } from '../hoc/ModalFormHOC';
import LoginFields from './FormFieldComponents';

export const RegisterForm = ({
  state, show, close, onSubmit, onChange,
}) => {
  const {
    errors,
    loading,
    open,
    user: { email, username, password },
  } = state;

  return (
    <div>
      <Link to="#" onClick={show}>
        Register
      </Link>
      <TransitionablePortal transition={modalOptions} onClose={close} open={open}>
        <Modal
          centered
          closeIcon
          id="modal-form"
          size="tiny"
          dimmer="blurring"
          open={open}
          onClose={close}
        >
          <Dimmer active={loading}>
            <Loader>Preparing your engagement</Loader>
          </Dimmer>
          <Modal.Header>Join the community</Modal.Header>
          { errors.randomError
            && (
            <Message negative>
              <Message.Header>{errors.randomError}</Message.Header>
            </Message>
            )}
          <Modal.Content>
            <Form className="innerForm " size="large" onSubmit={onSubmit}>
              {/* Rendering the form fields from the FormField component */}
              {LoginFields('text', 'username', onChange, username, 'Enter your username', 'Username:', errors.usernameError)}
              {LoginFields('email', 'email', onChange, email, 'example@gmail.com', 'Email:', errors.emailError)}
              {LoginFields('password', 'password', onChange, password, 'Enter your password', 'Password:', errors.passwordError)}
              <Button className="btn" type="submit">
                Register
              </Button>
            </Form>
          </Modal.Content>
          <Modal.Actions />
        </Modal>
      </TransitionablePortal>
    </div>
  );
};

RegisterForm.propTypes = {
  state: PropTypes.shape({
    user: PropTypes.shape({
      username: PropTypes.string.isRequired,
      email: PropTypes.string.isRequired,
      password: PropTypes.string.isRequired,
    }).isRequired,
    errors: PropTypes.shape({}).isRequired,
    loading: PropTypes.bool.isRequired,
    open: PropTypes.bool.isRequired,
  }).isRequired,
  show: PropTypes.func.isRequired,
  close: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default ModalFormHOC(RegisterForm);
