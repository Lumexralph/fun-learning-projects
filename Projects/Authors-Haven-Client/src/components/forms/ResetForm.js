import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import {
  Button,
  Loader,
  Dimmer,
  Message,
  Form,
} from 'semantic-ui-react';
import PropTypes from 'prop-types';
import FormField from './FormFieldComponents';
import Header from '../Header/HeaderComponent';
import Footer from '../FooterSlim';
import RenderDiv from '../../helpers/helper';

export class ResetForm extends Component {
  state = {
    user: {
      email: '',
      response: {},
    },
    loading: false,
  };

  componentDidUpdate(prevProps) {
    const { data, error } = this.props;
    if ((data !== prevProps.data) || (error !== prevProps.error)) {
      this.updateState(false);
    }
  }

  onSubmit = (event) => {
    event.preventDefault();
    const { user } = this.state;
    const { submit } = this.props;
    this.setState({
      loading: true,
    });
    submit(user.email);
  };

  onChange = (event) => {
    const { user } = this.state;
    this.setState({
      user: {
        ...user,
        [event.target.name]: event.target.value,
      },
    });
  };

  updateState(newName) {
    this.setState({
      loading: newName,
    });
  }

  render() {
    const { user, loading } = this.state;
    const { error, data } = this.props;
    return (
      <div id="reset-page">
        <Header text="Home" pathname="/login" />
        <div id="reset" className="reset-div">
          <Dimmer active={loading}><Loader>Sending</Loader></Dimmer>
          <h3>Enter your email to reset password&nbsp;&nbsp;</h3>
          {data ? <Message success id="result">{data}</Message> : null}
          {error && error.body ? <Message negative id="error">{error.body[0]}</Message> : null}
          {data ? <Redirect to="/email" /> : null}
          <Form onSubmit={this.onSubmit}>
            {FormField('email', 'email', this.onChange, user.email, 'Enter your email', '', '')}
            <Button type="submit" content="Continue" />
            <RenderDiv />
          </Form>
        </div>
        <Footer />
      </div>
    );
  }
}

ResetForm.propTypes = {
  submit: PropTypes.func.isRequired,
  error: PropTypes.string.isRequired,
  data: PropTypes.shape({
    data: PropTypes.shape({
      message: PropTypes.string.isRequired,
    }),
  }).isRequired,
};
export default ResetForm;
