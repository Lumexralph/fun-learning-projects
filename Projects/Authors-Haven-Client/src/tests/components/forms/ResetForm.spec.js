import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { ResetForm } from '../../../components/forms/ResetForm';
import ResetDiv from '../../../helpers/helper';


describe('RESET FORM', () => {
  const props = {
    submit: jest.fn(),
  };
  const sinonSpy = sinon.spy();
  const event = {
    preventDefault() {},
    target: {
      name: 'email',
      value: 'JakeJone@gmail.com',
    },
  };

  const wrapper = shallow(<ResetForm onSubmit={sinonSpy} {...props} />);
  it('Should call the onSubmit function', () => {
    wrapper.instance().onSubmit(event);
    expect(sinonSpy.calledWith()).toBe(false);
  });

  it('should call update state function', () => {
    const prevProp = wrapper.props();
    wrapper.find('Form').simulate('submit', { preventDefault() {} });
    wrapper.instance().updateState(true);
    wrapper.setProps({ data: 'data' });
    wrapper.instance().componentDidUpdate(prevProp);
  });

  it('Should call the on submit function and set the state', () => {
    wrapper.instance().onChange(event);
    const actual = wrapper.state().user.email;
    const expected = 'JakeJone@gmail.com';
    expect(actual).toBe(expected);
  });

  it('Should set state "loading" to true if called', () => {
    wrapper.instance().updateState(true);
    const actual = wrapper.state().loading;
    const expected = true;
    expect(actual).toBe(expected);
  });
  it('should render reset div properly', () => {
    const shallowWrapper = shallow(<ResetDiv />);
    expect(shallowWrapper.length).toBe(1);
    expect(shallowWrapper).toMatchSnapshot();
  });
});
