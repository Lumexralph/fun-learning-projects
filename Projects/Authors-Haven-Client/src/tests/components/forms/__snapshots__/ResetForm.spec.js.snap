// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RESET FORM should render reset div properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RenderDiv />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h4>
        Go back to
        <Link
          replace={false}
          to="/login"
        >
          Login
        </Link>
      </h4>,
      "className": "login",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Go back to",
          <Link
            replace={false}
            to="/login"
          >
            Login
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Go back to",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Login",
            "replace": false,
            "to": "/login",
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
      ],
      "type": "h4",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h4>
          Go back to
          <Link
            replace={false}
            to="/login"
          >
            Login
          </Link>
        </h4>,
        "className": "login",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Go back to",
            <Link
              replace={false}
              to="/login"
            >
              Login
            </Link>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Go back to",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Login",
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": "Login",
            "type": [Function],
          },
        ],
        "type": "h4",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
